src/components/list-item/list-item.tsx

```diff
@@ -147,44 +307,173 @@ export class ListItem implements ConditionalSlotComponent, InteractiveComponent
   }
 
   renderContentContainer(): VNode {
-    const { description, disabled, label, nonInteractive } = this;
+    const { description, disabled, label } = this;
     const hasCenterContent = !!label || !!description;
     const content = [this.renderContentStart(), this.renderContent(), this.renderContentEnd()];
 
-    return !nonInteractive ? (
-      <button
+    return (
+      <td
+        aria-label={label}
         class={{
           [CSS.contentContainer]: true,
-          [CSS.contentContainerButton]: true,
-          [CSS.hasCenterContent]: hasCenterContent
+          [CSS.hasCenterContent]: hasCenterContent,
+          [CSS.contentContainerDisabled]: disabled
         }}
-        disabled={disabled}
-        ref={(focusEl) => (this.focusEl = focusEl)}
-      >
-        {content}
-      </button>
-    ) : (
-      <div
-        class={{ [CSS.contentContainer]: true, [CSS.hasCenterContent]: hasCenterContent }}
-        ref={() => (this.focusEl = null)}
+        onClick={this.toggleSelected}
+        ref={(el) => (this.contentEl = el)}
+        role="gridcell"
       >
         {content}
-      </div>
+      </td>
     );
   }
 
   render(): VNode {
+    const {
+      openable,
+      open,
+      level,
+      setPosition,
+      setSize,
+      active,
+      label,
+      selected,
+      selectionAppearance,
+      selectionMode
+    } = this;
+
+    const showBorder = selectionMode !== "none" && selectionAppearance === "border";
+    const borderSelected = showBorder && selected;
+    const borderUnselected = showBorder && !selected;
+
     return (
-      <Host role="listitem">
-        <div class={CSS.container}>
+      <Host>
+        <tr
+          aria-expanded={openable ? toAriaBoolean(open) : null}
+          aria-label={label}
+          aria-level={level}
+          aria-posinset={setPosition}
+          aria-selected={toAriaBoolean(selected)}
+          aria-setsize={setSize}
+          class={{
+            [CSS.container]: true,
+            [CSS.containerBorderSelected]: borderSelected,
+            [CSS.containerBorderUnselected]: borderUnselected
+          }}
+          onFocus={this.focusCellNull}
+          onKeyDown={this.handleItemKeyDown}
+          ref={(el) => (this.containerEl = el)}
+          role="row"
+          tabIndex={active ? 0 : -1}
```

Comment:
We want the `aria-label` on the `td` since it gets focus. 

---

`-1` does allow click focusing. W3C specifies to use -1. https://www.w3.org/WAI/ARIA/apg/example-index/treegrid/treegrid-1